compose/config/config.py

```diff
@@ -464,12 +464,18 @@ def resolve_environment(service_dict):
         env.update(env_vars_from_file(env_file))
 
     env.update(parse_environment(service_dict.get('environment')))
-    return dict(resolve_env_var(k, v) for k, v in six.iteritems(env))
+    d = dict(resolve_env_var(k, v) for k, v in six.iteritems(env))
+    if '_' in d.keys():
+        del d['_']
+    return d
 
 
 def resolve_build_args(build):
     args = parse_build_arguments(build.get('args'))
-    return dict(resolve_env_var(k, v) for k, v in six.iteritems(args))
+    d = dict(resolve_env_var(k, v) for k, v in six.iteritems(args))
+    if '_' in d.keys():
+        del d['_']
+    return d
```

Comment:
Instead of using `'_'` as a special case, could we use the empty tuple (`()`).

If you `return ()` from `resolve_env_var()`, you can filter it here using:

```python
dict(filter(None, resolve_env_var(k, v) for k, v in six.iteritems(args)))
```